src/common/java/net/minecraftforge/gradle/common/util/EclipseHacks.java

```diff
@@ -20,89 +20,58 @@
-import java.nio.charset.StandardCharsets;
-import java.util.ArrayList;
-import java.util.HashMap;
 import java.util.HashSet;
-import java.util.List;
-import java.util.Map;
+import java.util.Iterator;
 import java.util.Set;
 
 public class EclipseHacks {
 
-    @SuppressWarnings({"unchecked", "UnstableApiUsage"})
     public static void doEclipseFixes(@Nonnull final MinecraftExtension minecraft, @Nonnull final ExtractNatives nativesTask, @Nonnull final DownloadAssets assetsTask, @Nonnull final TaskProvider<Task> makeSrcDirs) {
         final Project project = minecraft.getProject();
         final File natives = nativesTask.getOutput();
-        final File assets = assetsTask.getOutput();
+
+        final EclipseModel eclipseConv = (EclipseModel)project.getExtensions().getByName("eclipse");
+        final XmlFileContentMerger classpathMerger = eclipseConv.getClasspath().getFile();
 
         final String LIB_ATTR = "org.eclipse.jdt.launching.CLASSPATH_ATTR_LIBRARY_PATH_ENTRY";
 
         project.getTasks().withType(GenerateEclipseClasspath.class, task -> {
             task.dependsOn(nativesTask, assetsTask, makeSrcDirs.get());
+        });
 
-            task.doLast(t -> {
-                try {
-                    Node xml = new XmlParser().parse(task.getOutputFile());
-
-                    List<Node> entries = (ArrayList<Node>)xml.get("classpathentry");
-                    Set<String> paths = new HashSet<>();
-                    List<Node> remove = new ArrayList<>();
-
-                    entries.stream().filter(e -> "src".equals(e.get("@kind"))).forEach(e -> {
-                        if (!paths.add((String)e.get("@path"))) { //Eclipse likes to duplicate things... No idea why, lets kill them off
-                            remove.add(e);
-                        }
-
-                        if (((List<Node>)e.get("attributes")).isEmpty()) {
-                            e.appendNode("attributes");
-                        }
-
-                        Node attr = ((List<Node>)e.get("attributes")).get(0);
-
-                        if (((List<Node>)attr.get("attribute")).stream().noneMatch(n -> LIB_ATTR.equals(n.get("@name")))) {
-                            attr.appendNode("attribute", props("name", LIB_ATTR, "value", natives.getAbsolutePath()));
-                        }
-                    });
-
-                    remove.forEach(xml::remove);
+        classpathMerger.whenMerged(obj -> {
+            Classpath classpath = (Classpath)obj;
+            Set<String> paths = new HashSet<>();
+            Iterator<ClasspathEntry> itr = classpath.getEntries().iterator();
+            while (itr.hasNext()) {
+                ClasspathEntry entry = itr.next();
+                if (entry instanceof SourceFolder) {
+                    SourceFolder sf = (SourceFolder)entry; 
+                    if (!paths.add(sf.getPath())) {
+                        //Eclipse likes to duplicate things... No idea why, let's kill them off
```

Comment:
This probably has to deal with making changes in the IDE, and then because one version does not match exactly, it recreates the original version in addition.  But I'm not 100% sure of that.  Removing them is probably still fine.